src/main/java/com/google/sps/servlets/getData.java

```diff
@@ -55,133 +45,91 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 // #######################
 // # Retrieving the Data #
 // #######################
+
+   // debug long startingTime = System.currentTimeMillis();  avg time 1.35 seconds first attempt, .5 seconds for second attempt
       
```

Comment:
Here's where a few changes start to begin, for example, I learn that I don't need a URLConnection object to read from the web. Nice! Also, I change from scanner to BufferedReader because I realized I wasn't really using the scanner to its full potential, and I only needed to read lines.